content/en/blog/_posts/2023-11-xx-scheduler-queueinghint/index.md

```diff
@@ -0,0 +1,99 @@
+date: 2023-10-25T10:00:00-08:00
+slug: scheduler-queueinghint
+---
+
+**Author:** [Kensei Nakada](https://github.com/sanposhiho) (Mercari)
+
+The scheduler is the core component that decides which Node Pods run on.
+Basically, it schedules Pods **one by one**, 
+and thus the larger your cluster is, the more crucial the throughput of the scheduler is.
+
+The throughput of the scheduler is our eternal challenge, 
+over the years, SIG-Scheduling have been putting effort to improve the scheduling throughput by many enhancements.
+
+In this blog post, I'll introduce a recent major improvement in the scheduler, named QueueingHint.
+
+We'll go through the explanation of the basic background knowledge of the scheduler,
+and how QueueingHint improves our scheduling throughput.
+
+## Scheduling Queue
+
+The scheduler has Scheduling Queue which has all unscheduled Pods.
+
+Scheduling Queue is composed of three places in it - ActiveQ, BackoffQ and Unschedulable Pod Pool.
+- ActiveQ: Pods which are ready to get scheduling.
+- BackoffQ: Pods which are waiting for the backoff, and will be put into ActiveQ after that.
+- Unschedulable Pod Pool: Pods which should not be scheduled for now.
+
+## Scheduling Framework and Plugins
+
+[Scheduling Framework](/docs/concepts/scheduling-eviction/scheduling-framework/)
+
+The scheduler is implemented with Scheduling Framework. 
+And, each scheduling requirements/preferences is implemented as a plugin.
+(e.g., PodAffinity is implemented in the PodAffinity plugin.)
+
+The first phase, called Scheduling Cycle, takes Pods from activeQ **one by one**, gather all plugins' idea,
+and lastly decides a Node to run the Pod, or concludes that the Pod cannot go to anywhere for now.
+
+If the scheduling is successful, the second phase, called Binding Cycle, binds the Pod with the Node.
+But, if it turns out that the Pod cannot go to anywhere in Scheduling Cycle, 
+Binding Cycle isn't executed, instead the Pod is moved back to Scheduling Queue.
+There are some exception cases though, such unscheduled Pod is basically put into Unschedulable Pod Pool.
+
+Pods in Unschedulable Pod Pool are moved to ActiveQ/BackoffQ 
+only when Scheduling Queue thinks they might be schedulable if we retry the scheduling.
+
+That is a crucial step because Scheduling Cycle is performed for Pods one by one -
+if we didn't have Unschedulable Pod Pool and kept retrying the scheduling of any Pods, 
+Scheduling Cycle is wasted for Pods with no hope to be scheduled.
+
+Then, how do they decide when to move? How do they notice that Pods might be schedulable now?
+There we go, QueueingHint comes in.
+
+## QueueingHint
+
+QueueingHint is callback functions per plugin to notice an object addition/update/deletion in the cluster (we call them cluster events)
+that may make Pods schedulable.
+
+Let's say PodA has a required PodAffinity, and got rejected in scheduling cycle by PodAffinity plugin 
+because no Node has any Pod matching with PodA's PodAffinity.
```

Comment:
```suggestion
Let's say the Pod `pod-a` has a required Pod affinity, and got rejected in scheduling cycle by the `PodAffinity` plugin
because no Node has any Pod matching the Pod affinity specification for `pod-a`.
```